### PR TITLE
[libc++][ranges] LWG3618: Unnecessary `iter_move` for `transform_view::iterator`

### PR DESCRIPTION
**Pull request by xiaoyang-sde**
## Introduction

This patch implements LWG3618: Unnecessary `iter_move` for `transform_view::iterator`.

`transform_view`'s iterator currently specifies a customization point for `iter_move`. This customization point does the same thing that the default implementation would do, but its sole purpose is to ensure the appropriate conditional `noexcept` specification.

## Reference

- [[range.transform.iterator]](https://eel.is/c++draft/range.transform.iterator)
- [LWG3618](https://cplusplus.github.io/LWG/issue3618)

### CODE REVIEW

File: libcxx/docs/Status/Cxx23Issues.csv [hunk 1]
```diff
@@ -146,7 +146,7 @@
 "`3610 <https://wg21.link/LWG3610>`__","``iota_view::size`` sometimes rejects integer-class types","February 2022","","","|ranges|"
 "`3612 <https://wg21.link/LWG3612>`__","Inconsistent pointer alignment in ``std::format`` ","February 2022","|Complete|","14.0","|format|"
 "`3616 <https://wg21.link/LWG3616>`__","LWG 3498 seems to miss the non-member ``swap`` for ``basic_syncbuf`` ","February 2022","",""
-"`3618 <https://wg21.link/LWG3618>`__","Unnecessary ``iter_move`` for ``transform_view::iterator`` ","February 2022","","","|ranges|"
+"`3618 <https://wg21.link/LWG3618>`__","Unnecessary ``iter_move`` for ``transform_view::iterator`` ","February 2022","|Complete|","19.0","|ranges|"
```
Comment (mordante) [inline libcxx/docs/Status/Cxx23Issues.csv:149]: The LWG-issues adds a `noexcept`. This is not tested in `libcxx/test/std/ranges/range.adaptors/range.transform/iterator/deref.pass.cpp`. Please update this test.

**Review by ldionne — CHANGES_REQUESTED** [review]
Gentle ping, it looks like there is outstanding feedback that needs to be addressed.

File: libcxx/test/std/ranges/range.adaptors/range.transform/iterator/deref.pass.cpp [hunk 2]
```diff
@@ -37,7 +37,7 @@ int main(int, char**) {
     using View = std::ranges::transform_view<MoveOnlyView, PlusOneNoexcept>;
     View transformView(MoveOnlyView{buff}, PlusOneNoexcept{});
     assert(*transformView.begin() == 1);
-    LIBCPP_ASSERT_NOEXCEPT(*std::declval<std::ranges::iterator_t<View>>());
+    ASSERT_NOEXCEPT(*std::declval<std::ranges::iterator_t<View>>());
```
Comment (xiaoyang-sde) [inline libcxx/test/std/ranges/range.adaptors/range.transform/iterator/deref.pass.cpp:40]: The `operator*` previously provided a `noexcept` specifier as a libc++ extension. Following the adoption of this LWG issue, the `noexcept` specifier is now standard behavior.

**Review by ldionne — APPROVED** [review]
Thanks for clarifying with your comment. LGTM!

**Comment by xiaoyang-sde** [comment]
> Thanks for clarifying with your comment. LGTM!

Could you please merge this patch? Thank you!

**Pull request merged**